### PR TITLE
Move gRPC library installation into Makefile and link it statically

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Comment by GuanLuo** [comment]
Yes, I think libprotobuf can also be linked statically.

When you say show Makefile build steps, you mean to mention what the Makefile is doing (i.e. building gRPC library, building client library, building clients..) and what are the dependencies that the user's local system should meet for each of them?

File: README.md [hunk 1]
```diff
@@ -48,11 +48,41 @@ is also available for inference server issues and questions.
...
+Before building the client libraries and applications you must first
+install some prerequisites. The following instructions assume Ubuntu
+16.04. OpenCV is used by image\_client to preprocess images before
+sending them to the inference server for inferencing. The python-pil
+package is required by the Python image\_client example.
+
+    sudo apt-get update
+    sudo apt-get install build-essential libcurl3-dev libopencv-dev libopencv-core-dev python-pil software-properties-common
+
+Protobuf3 support and gRPC support are required, they will be
```
Comment (deadeyegoodwin) [inline README.md]: Delete this paragraph and merge the install packages here with those above.

File: README.md [hunk 2]
```diff
@@ -48,11 +48,41 @@ is also available for inference server issues and questions.
...
+    sudo apt-get update
+    sudo apt-get install autoconf automake libtool pkg-config
+    
+Creating the whl file for the Python client library requires setuptools. 
+And grpcio-tools is required for gRPC support in Python client library.
+
+    pip install --no-cache-dir --upgrade setuptools grpcio-tools
+
+With those prerequisites installed, the C++ and Python client libraries
+and example image\_client application can be built:
```
Comment (deadeyegoodwin) [inline README.md]: ... image\_client and perf\_client applications...

File: README.md [hunk 3]
```diff
@@ -48,11 +48,35 @@ is also available for inference server issues and questions.
...
+    
+Creating the whl file for the Python client library requires setuptools. 
+And grpcio-tools is required for gRPC support in Python client library.
+
+    pip install --no-cache-dir --upgrade setuptools grpcio-tools
+
+With those prerequisites installed, the C++ and Python client libraries
+and example image\_client and perf\_client applications can be built:
+
+    sudo make -f Makefile.clients all pip
```
Comment (deadeyegoodwin) [inline README.md]: Why do we need sudo on the make?

Comment (GuanLuo) [reply README.md]: it is needed for installing gRPC

**Pull request merged**